File: ssTools/netcat/ruby/nc.rb
require 'socket'

def handle_connection(client)
    puts "Menerima koneksi dari #{client.peeraddr[2]}"

    loop do
    data = client.recv(1024)
    break if data.empty?

    puts data
    end

    client.close
end

server = TCPServer.new(6969)
puts 'Menjalankan server netcat di port 6969...'

loop do
    client = server.accept
    Thread.new { handle_connection(client) }
end

File: ssTools/netcat/python/nc.py
import socket

host = "127.0.0.1"  
port = 6969

server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

server_socket.bind((host, port))

server_socket.listen(1)
print("Menjalankan server netcat di port 6969...")

client_socket, client_address = server_socket.accept()
print("Terhubung dari:", client_address)

while True:
    data = client_socket.recv(1024).decode()
    if not data:
        break

    print(data)

# Menutup koneksi
client_socket.close()
server_socket.close()
